aws/resource_aws_eip_association.go

```diff
@@ -157,7 +157,30 @@ func resourceAwsEipAssociationRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	response, err := conn.DescribeAddresses(request)
+	var response *ec2.DescribeAddressesOutput
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		response, err = conn.DescribeAddresses(request)
+
+		if tfawserr.ErrCodeEquals(err, "InvalidAssociationID.NotFound") {
+			return resource.RetryableError(err)
+		}
+
+		if d.IsNewResource() && (response.Addresses == nil || len(response.Addresses) == 0) {
+			return resource.RetryableError(&resource.NotFoundError{})
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		response, err = conn.DescribeAddresses(request)
+	}
```

Comment:
Before returning it as an error below, it seems like we should also remove from state to trigger recreation if it is not a new resource.

```suggestion
	if isResourceTimeoutError(err) {
		response, err = conn.DescribeAddresses(request)
	}

	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, "InvalidAssociationID.NotFound") {
		log.Printf("[WARN] EIP Association (%s) not found, removing from state", d.Id())
		d.SetId("")
		return nil
	}
```